src/nodeTypes.ml

```diff
@@ -0,0 +1,28 @@
+module ArrayBuffer = struct
+  type t
+end
```

Comment:
Not sure why you're doing it like this, but having the types wrapped in a module is going to run you into trouble eventually, when you need dependencies between them. Mutual recursion between modules is a mess. Dependencies between types in the same file, or from modules to types in a separate module is very straight forward.

Also not sure why you have an ArrayBuffer with a type t here, and then another ArrayBuffer module as a separate file with it's own type t. They're going to be treated as completely separate types.